Subject: [PATCH] m68k: Don't include RODATA into text segment

From: Roman Zippel <zippel@linux-m68k.org>

Don't include RODATA into text segment as it includes the kallsyms data
and can cause spurious link failures (layout differences can change the
number of symbols in kallsym, i.e. when a symbol is equal to _etext it's
not included).

Signed-off-by: Roman Zippel <zippel@linux-m68k.org>
Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>

---
 arch/m68k/kernel/vmlinux-std.lds |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/arch/m68k/kernel/vmlinux-std.lds
+++ b/arch/m68k/kernel/vmlinux-std.lds
@@ -18,6 +18,8 @@ SECTIONS
 	*(.gnu.warning)
 	} :text = 0x4e75
 
+  _etext = .;			/* End of text section */
+
   . = ALIGN(16);		/* Exception table */
   __start___ex_table = .;
   __ex_table : { *(__ex_table) }
@@ -25,8 +27,6 @@ SECTIONS
 
   RODATA
 
-  _etext = .;			/* End of text section */
-
   .data : {			/* Data */
 	DATA_DATA
 	CONSTRUCTORS
